packages/lib/services/searchengine/SearchEngine.ts

```diff
@@ -554,6 +554,94 @@ export default class SearchEngine {
 		return SearchEngine.SEARCH_TYPE_FTS;
 	}
 
+	makeSearchTree(str: string) {
+		// (tag:A (any:1 tag:B tag:C)) -> [ "tag:A", [ "any:1", "tag:B", "tag:C"]]
+		let i = 0;
+		function main(): any {
+			const arr = [];
+			let startIndex = i;
+			function addWord() {
+				if (i - 1 > startIndex) {
+					arr.push(str.slice(startIndex, i - 1));
+				}
+			}
+			while (i < str.length) {
+				switch (str[i++]) {
+				case ' ':
+					addWord();
+					startIndex = i;
+					continue;
+				case '(':
+					arr.push(main());
+					startIndex = i;
+					continue;
+				case ')':
+					addWord();
+					return arr;
+				}
+			}
+			addWord();
+			return arr;
+		}
+		return main();
+	}
+
+
+
+	async makeQuery(parseTree: any) {
+
+		const getQueryWithParms = async (searchString: string) => {
+			const parsedQuery = await this.parseQuery(searchString);
+			const { query, params } = queryBuilder(parsedQuery.allTerms);
+			return { query, params };
+		};
+
+		const intersection = (queries: any) => {
```

Comment:
Likewise please use an interface instead of "any". I guess the original code was in JS so maybe we have no interface for all this, but it makes sense to create one now and to start using it for new code. Also please set the return type.